File: output/CodeGen/avatar/Java/Python/codeforces_456_A.py
def main():
    n = int(input())
    a = [int(i) for i in input().split()]
    b = [int(i) for i in input().split()]
    temp = 0
    q = 0
    w = 0
    e = 0
    f = False
    for i in range(n):
        if temp < a[i]:
            if q > b[i]:
                f = True
            q = b[i]
            temp = a[i]
        if temp > a[i]:
            if q < b[i]:
                f = True
            q = b[i]
            temp = a[i]
        if a[i] > w:
            w = a[i]
            if b[i] < e:
                f = True
            e = b[i]
        if a[i] < w:
            if b[i] > e:
                f = True
    if f == True:
        print("Happy Alex")
    else:
        print("Poor Alex")

if __name__ == '__main__':
    main()

File: output/CodeGen/avatar/Python/Java_handraft/codeforces_456_A.java
import java.util.*;

public class codeforces_456_A {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[][] m = new int[n][2];
        for(int i = 0; i < n; i++){
            String[] mItem = in.nextLine().split(" ");
            m[i][0] = Integer.parseInt(mItem[0]);
            m[i][1] = Integer.parseInt(mItem[1]);
        }
        m = sort(m, true);
        int a = m[0][0];
        for(int i = 1; i < n; i++){
            if(m[i][1] > a){
                System.out.println("Happy Alex");
                break;
            }
            a = m[i][0];
        }
        if(a == m[n-1][0]){
            System.out.println("Poor Alex");
        }
    }
    public static int[][] sort(int[][] m, boolean reverse){
        int n = m.length;
        for(int i = 0; i < n; i++){
            for(int j = i+1; j < n; j++){
                if(m[i][1] > m[j][1]){
                    int temp = m[i][0];
                    m[i][0] = m[j][0];
                    m[j][0] = temp;
                }
            }
        }
        if(reverse){
            for(int i = 0; i < n; i++){
                for(int j = 0; j < n; j++){
                    m[i][j] = m[i][j] * -1;
                }
            }
        }
        return m;
    }
}